docs/src/modules/java/pages/eventsourced.adoc

```diff
@@ -3,108 +3,137 @@
+== Registering the Entity
 
-Event handlers may be declared for a superclass or interface based on type. For example, an event handler that declares an `Object` parameter handles all events. In the case where multiple event handlers match, Akka Serverless chooses the most specific event handler from all superclass interfaces.
+To make Akka Serverless aware of the Event Sourced Entity, we need to register it with the service.
 
-The following example shows an event handler for the `ItemAdded` event with a utility method, `convert`:
+From the code-generation, the registration gets automatically inserted in the generated `AkkaServerlessFactory.withComponents` method from the `Main` class.
 
-[source,java,indent=0]
+[source,java]
+.src/main/java/com/example/shoppingcart/Main.java
 ----
-include::example$java-eventing-shopping-cart/src/main/java/shopping/cart/domain/ShoppingCartEntity.java[tag=itemAdded]
+include::example$java-eventsourced-shopping-cart/src/main/java/com/example/shoppingcart/Main.java[]
 ----
 
-== Snapshots
+By default, the generated constructor has a `EventSourcedEntityContext` parameter, but you can change this to accept other parameters. If you change the constructor of the `ShoppingCart` class you will see a compilation error here, and you have to adjust the factory function that is passed to `AkkaServerlessFactory.withComponents`.
 
-Snapshots are an important optimization for Event Sourced Entities that emit many events. Rather than reading the entire journal upon loading or restart, Akka Serverless can initiate them from a snapshot.
+When more components are added the `AkkaServerlessFactory` is regenerated and you have to adjust the registration from the `Main` class.
 
-Snapshots are handled automatically by Akka Serverless without any specific code required. By default, a snapshot is taken for every 100:th event emitted. Tuning how often a snapshot of the state is taken can be configured using config or programmatically handing a `EventSourcedEntityOptions` to the provider when registering the entity.
+== Snapshots
 
-== Registering the Entity
+Snapshots are an important optimization for Event Sourced Entities that emit many events. Rather than reading the entire journal upon loading or restart, Akka Serverless can initiate them from a snapshot.
 
-Once you've created your Entity, you can register it with the link:{attachmentsdir}/api/com/akkaserverless/javasdk/AkkaServerless.html[`AkkaServerless` {tab-icon}, window="new"] server by invoking the `registerEventSourcedEntity` method. In addition to passing your Event Sourced Entity class and protobuf service descriptor, you also need to pass any descriptors that you use for persisting events, for example, `ShoppingCart` as defined in the `shopping-cart-domain.proto` descriptor.
+Snapshots are stored and handled automatically by Akka Serverless without any specific code required. Snapshots are stored after a configured number of events:
```

Comment:
Should we mention the actual default value? (every 100 events)

Here we describe the config based way which is global to all event sourced entities in the project, should we mention the programmatic/per entity type way as well? (Also, should we recommend one over the other?)

Reply:
I actually implemented a snippet for the programatic but then though it was much easier to use config. Since an Akka Serverless Service typically has only one entity type the config should work in most cases.

> Should we mention the actual default value? (every 100 events)

I'm a bit allergic to mentioning default values in docs. The snippet there shows the config with 100.

If we want more in depth reference docs we should show (parts of) reference.conf.

Reply:
If I put my user hat on, I think I'd find it somewhat important to know the the default, so perhaps worth pulling in reference.conf then (or at least mentioning where they can find the default value)?